airflow/providers/slack/hooks/slack.py

```diff
@@ -45,28 +65,124 @@ class SlackHook(BaseHook):
+            warnings.warn("`use_session` has no affect in slack_sdk.WebClient.", UserWarning, stacklevel=2)
+
+        self._token = token
+        if self._token:
+            warnings.warn(
+                "Provide token as hook argument deprecated by security reason and will be removed "
+                "in a future releases. Please specify token in `Slack API` connection.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+
+    @cached_property
+    def client(self) -> WebClient:
+        """Get the underlying slack_sdk.WebClient (cached)."""
+        return WebClient(**self._get_conn_params())
+
+    def get_conn(self) -> WebClient:
+        """Get the underlying slack_sdk.WebClient (cached)."""
+        return self.client
+
+    def _get_conn_params(self) -> Dict[str, Any]:
+        """Fetch connection params as a dict and merge it with hook parameters."""
+        conn = self.get_connection(self.slack_conn_id) if self.slack_conn_id else None
+        conn_params: Dict[str, Any] = {}
+
+        if self._token:
+            conn_params["token"] = self._token
+        elif conn:
+            if not conn.password:
+                raise AirflowNotFoundException(
+                    f"Connection ID {self.slack_conn_id!r} does not contain password (Slack API Token)."
+                )
+            conn_params["token"] = conn.password
+
+        extra_config = ConnectionExtraConfig(
+            conn_type=self.conn_type,
+            conn_id=conn.conn_id if conn else None,
+            extra=conn.extra_dejson if conn else {},
+        )
+
+        # Merge Hook parameters with Connection config
+        conn_params.update(
+            {
+                "timeout": self.timeout or extra_config.getint("timeout", default=None),
+                "base_url": self.base_url or extra_config.get("base_url", default=None),
+                "proxy": self.proxy or extra_config.get("proxy", default=None),
+                "retry_handlers": (
+                    self.retry_handlers or extra_config.getimports("retry_handlers", default=None)
+                ),
+            }
+        )
+
+        # Add additional client args
+        conn_params.update(self.extra_client_args)
+        if "logger" not in conn_params:
+            conn_params["logger"] = self.log
+
+        return {k: v for k, v in conn_params.items() if v is not None}
+
+    @cached_property
+    def token(self) -> str:
+        warnings.warn(
+            "`SlackHook.token` property deprecated and will be removed in a future releases.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        return self._get_conn_params()["token"]
 
     def __get_token(self, token: Any, slack_conn_id: Any) -> str:
+        warnings.warn(
+            "`SlackHook.__get_token` method deprecated and will be removed in a future releases.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
Also might be better just remove this method.
It not used after changes and it is hardly possible that someone override it